modules/programs/tmux.nix

```diff
@@ -0,0 +1,203 @@
+        description = "Path of the configuration file to include.";
+      };
+
+      extraConfig = mkOption {
+        type = types.lines;
+        description = "Additional configuration for the associated plugin.";
+        default = "";
+      };
+    };
+  };
+in
+{
+  options = {
+
+    programs.tmux = {
+      enable = mkEnableOption "Tmux";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.tmux;
+        defaultText = "pkgs.tmux";
+        example = literalExample "pkgs.tmux";
+        description = "tmux";
+      };
+
+      sensibleOnTop = mkOption {
+        type = types.bool;
+        default = false;
+        description = "run the sensible plugin at the top of your .conf (to overwrite configs)";
+      };
+
+      tmuxp = mkOption {
+        type = tmuxpModule;
+        default = {};
+        description = "Options to configure Tmuxp";
+      };
+
+      tmuxinator = mkOption {
+        type = tmuxinatorModule;
+        default = {};
+        description = "Options to configure Tmuxinator";
+      };
+
+      plugins = mkOption {
+        type = types.listOf (types.either types.package pluginModule);
+        description = ''
+          List of tmux plugins to include in your .tmux.conf
+            -- defaults to the sensible plugin.
+        '';
+        default = [ ];
+        example = literalExample ''
+        with pkgs; [
+          tmuxPlugins.sensible
+          {
+            plugin = tmuxPlugins.resurrect;
+            extraConfig = "set -g @resurrect-strategy-nvim 'session'";
+          }
+          {
+            plugin = tmuxPlugins.continuum;
+            extraConfig = ''''''
+              set -g @continuum-restore 'on'
+              set -g @continuum-save-interval '60' # minutes
+            '''''';
+          }
+        ];
+        '';
+      };
+
+      extraConfig = mkOption {
+        type = types.lines;
+        description = "Additional configuration to add.";
+        default = "";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable (
+    mkMerge [
+      {
+        home.packages = [ cfg.package ]
+          ++ optional cfg.tmuxinator.enable cfg.tmuxinator.package
+          ++ optional cfg.tmuxp.enable      cfg.tmuxp.package;
+      }
+
+      (mkIf (cfg.extraConfig != "") {
+        home.file.".tmux.conf".text = cfg.extraConfig;
+      })
+
+      (mkIf (cfg.sensibleOnTop) {
+        home.file.".tmux.conf".text = mkBefore ''
+        # ============================================= #
+        # Start with defaults from the Sensible plugin  #
+        # --------------------------------------------- #
+        run-shell ${pkgs.tmuxPlugins.sensible.rtp}
+        # ============================================= #
+        '';
+      })
+      (mkIf (cfg.plugins != []) {
+        assertions = [
+          (let
+            not = x: if x then false else true;
+            badPlugins = filter (p: not (hasPrefix "tmuxplugin" (pluginName p))) cfg.plugins;
```

Comment:
This conditional is a little brittle. Tmux plugins are all prefixed this way, but may not be named like this in the future. A better condition would be checking if the package is in the `pkgs.tmuxPlugins.*` namespace, but I wasn't sure how to do this.

Reply:
Could use `hasAttr (pluginName p) pkgs.tmuxPlugins`?